tfe/resource_tfe_project_policy_set.go

```diff
@@ -0,0 +1,173 @@
+
+	policySetAddProjectsOptions := tfe.PolicySetAddProjectsOptions{}
+	policySetAddProjectsOptions.Projects = append(policySetAddProjectsOptions.Projects, &tfe.Project{ID: projectID})
+
+	err := config.Client.PolicySets.AddProjects(ctx, policySetID, policySetAddProjectsOptions)
+	if err != nil {
+		return fmt.Errorf(
+			"Error attaching policy set id %s to project %s: %w", policySetID, projectID, err)
+	}
+
+	d.SetId(fmt.Sprintf("%s_%s", projectID, policySetID))
+
+	return resourceTFEProjectPolicySetRead(d, meta)
+}
+
+func resourceTFEProjectPolicySetRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	log.Printf("[DEBUG] Read configuration of project policy set: %s", policySetID)
+	policySet, err := config.Client.PolicySets.ReadWithOptions(ctx, policySetID, &tfe.PolicySetReadOptions{
+		Include: []tfe.PolicySetIncludeOpt{tfe.PolicySetProjects},
+	})
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Policy set %s no longer exists", policySetID)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading configuration of policy set %s: %w", policySetID, err)
+	}
+
+	isProjectAttached := false
+	for _, project := range policySet.Projects {
+		if project.ID == projectID {
+			isProjectAttached = true
+			d.Set("project_id", projectID)
+			break
+		}
+	}
+
+	if !isProjectAttached {
+		log.Printf("[DEBUG] Project %s not attached to policy set %s. Removing from state.", projectID, policySetID)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("policy_set_id", policySetID)
+	return nil
+}
+
+func resourceTFEProjectPolicySetDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	policySetID := d.Get("policy_set_id").(string)
+	projectID := d.Get("project_id").(string)
+
+	log.Printf("[DEBUG] Detaching project (%s) from policy set (%s)", projectID, policySetID)
+	policySetRemoveProjectsOptions := tfe.PolicySetRemoveProjectsOptions{}
+	policySetRemoveProjectsOptions.Projects = append(policySetRemoveProjectsOptions.Projects, &tfe.Project{ID: projectID})
+
+	err := config.Client.PolicySets.RemoveProjects(ctx, policySetID, policySetRemoveProjectsOptions)
+	if err != nil {
+		return fmt.Errorf(
+			"Error detaching project %s from policy set %s: %w", projectID, policySetID, err)
+	}
+
+	return nil
+}
+
+func resourceTFEProjectPolicySetImporter(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	// The format of the import ID is <ORGANIZATION/PROJECT ID/POLICYSET NAME>
+	splitID := strings.SplitN(d.Id(), "/", 3)
+	if len(splitID) != 3 {
+		return nil, fmt.Errorf(
+			"invalid project policy set input format: %s (expected <ORGANIZATION>/<PROJECT ID>/<POLICYSET NAME>)",
+			splitID,
+		)
+	}
+
+	organization, projectID, pSName := splitID[0], splitID[1], splitID[2]
```

Comment:
Good work! 
Just a minor nit: I'd like to advocate for clear var names like `policySetName` instead of `pSName` 💅

Reply:
same! good call @laurenolivia